hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -310,35 +256,159 @@ private void doRefresh() {
             )
         );
 
-    cachedFileSize = cachedAllInputFileSlices.values().stream()
+    this.cachedFileSize += ret.values().stream()
         .flatMap(Collection::stream)
         .mapToLong(BaseHoodieTableFileIndex::fileSliceSize)
         .sum();
 
-    // If the partition value contains InternalRow.empty, we query it as a non-partitioned table.
-    queryAsNonePartitionedTable = partitionFiles.keySet().stream().anyMatch(p -> p.values.length == 0);
+    return ret;
+  }
 
-    long duration = System.currentTimeMillis() - startTime;
+  /**
+   * Get partition path with the given partition value
+   * @param partitionNames partition names
+   * @param values partition values
+   * @return partitions that match the given partition values
+   */
+  protected List<PartitionPath> getPartitionPaths(String[] partitionNames, String[] values) {
+    if (cachedAllPartitionPaths != null) {
+      LOG.info("All partition paths have already loaded, use it directly");
+      return cachedAllPartitionPaths;
+    }
 
-    LOG.info(String.format("Refresh table %s, spent: %d ms", metaClient.getTableConfig().getTableName(), duration));
+    Pair<String, Boolean> relativeQueryPartitionPathPair = composeRelativePartitionPaths(partitionNames, values);
```

Comment:
We actually don't need to return boolean flag, we can detect whether this would be a full-path here as well (comparing # of values against # of partition columns)